app/src/main/java/org/apache/taverna/mobile/ui/tutorial/TutorialActivity.java

```diff
@@ -0,0 +1,204 @@
+            getWindow().getDecorView()
+                    .setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+        }
+
+        layouts = new int[]{
+                R.layout.tutorial_slide1,
+                R.layout.tutorial_slide2,
+                R.layout.tutorial_slide3,
+                R.layout.tutorial_slide4};
+
+        addBottomDots(0);
+
+        viewPagerAdapter = new ViewPagerAdapter();
+        viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(viewPagerPageChangeListener);
+
+    }
+
+    @OnClick(R.id.btn_skip)
+    public void skipClick(View v) {
+        launchFlashScreen();
+    }
+
+    @OnClick(R.id.btn_next)
+    public void nextClick(View v) {
+        int current = getItem(+1);
+        if (current < layouts.length) {
+            viewPager.setCurrentItem(current);
+        } else {
+            launchFlashScreen();
+        }
+    }
+
+    public void addBottomDots(int currentPage) {
+        dots = new TextView[layouts.length];
+
+        int[] colorsActive = getResources().getIntArray(R.array.array_dot_active);
+        int[] colorsInactive = getResources().getIntArray(R.array.array_dot_inactive);
+
+        dotsLayout.removeAllViews();
+        for (int i = 0; i < dots.length; i++) {
+            dots[i] = new TextView(this);
+            dots[i].setText(Html.fromHtml("&#8226;"));
+            dots[i].setTextSize(35);
+            dots[i].setTextColor(colorsInactive[currentPage]);
+            dotsLayout.addView(dots[i]);
+        }
+
+        if (dots.length > 0)
+            dots[currentPage].setTextColor(colorsActive[currentPage]);
+    }
+
+    public int getItem(int i) {
+        return viewPager.getCurrentItem() + i;
+    }
+
+    public void launchFlashScreen() {
+        preferencesHelper.setFirstTimeLaunch(false);
+        startActivity(new Intent(TutorialActivity.this, FlashScreenActivity.class));
+        finish();
+    }
+
+    ViewPager.OnPageChangeListener viewPagerPageChangeListener =
+            new ViewPager.OnPageChangeListener() {
+
+                @Override
+                public void onPageSelected(int position) {
+                    addBottomDots(position);
+
+                    if (position == layouts.length - 1) {
+                        bNext.setText(getString(R.string.start));
+                        bSkip.setVisibility(View.GONE);
+                    } else {
+                        bNext.setText(getString(R.string.next));
+                        bSkip.setVisibility(View.VISIBLE);
+                    }
+                }
+
+                @Override
+                public void onPageScrolled(int arg0, float arg1, int arg2) {
+
+                }
+
+                @Override
+                public void onPageScrollStateChanged(int arg0) {
+
+                }
+            };
+
+    public class ViewPagerAdapter extends PagerAdapter {
```

Comment:
Move it to separate class.

Reply:
@sagar15795 layouts position will not be reachable by  moving it in separate class. If there is any other way then do let me know. 👍

Reply:
https://www.bignerdranch.com/blog/viewpager-without-fragments/

This link might help.